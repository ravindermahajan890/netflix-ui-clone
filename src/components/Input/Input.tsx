import { InputAdornment, InputBase, InputLabel, InputLabelProps, styled } from '@mui/material';
import { ChangeEvent, ReactElement } from 'react';

const CustomInputLabel = styled(InputLabel)<InputLabelProps>(() => ({
  color: 'white',
  fontSize: '20px',
}));

const CustomInputBox = styled(InputBase)(({ theme }) => ({
  border: `1px solid ${theme.palette.lightGrey.main}`,
  borderRadius: '4px',
  width: '100%',
  padding: theme.spacing(2),
  marginTop: theme.spacing(1),
  '& .MuiInputBase-input': { fontSize: '20px', fontWeight: '300', padding: '0' },
}));

interface InputProps {
  label: string;
  icon: ReactElement;
  type?: string;
  onChange?: (event: ChangeEvent) => void;
}

const Input = ({ label, icon, type, onChange }: InputProps) => (
  <div>
    <CustomInputLabel>{label}</CustomInputLabel>
    <CustomInputBox
      startAdornment={<InputAdornment position="start">{icon}</InputAdornment>}
      type={type}
      onChange={onChange}
    />
  </div>
);

export default Input;
