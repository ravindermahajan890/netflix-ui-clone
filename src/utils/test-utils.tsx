// eslint-disable-next-line import/no-extraneous-dependencies
import { render as rtlRender } from '@testing-library/react';
import { PropsWithChildren, ReactElement } from 'react';
import { HashRouter } from 'react-router-dom';
import Providers from '../providers';

// eslint-disable-next-line import/prefer-default-export
export const render = (ui: ReactElement) => {
  const wrapper = ({ children }: PropsWithChildren) => (
    <Providers>
      <HashRouter>{children}</HashRouter>
    </Providers>
  );

  return rtlRender(ui, { wrapper });
};
