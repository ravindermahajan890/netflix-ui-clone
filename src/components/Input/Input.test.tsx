import { render } from '../../utils/test-utils';
import Input from '.';

describe('Input', () => {
  it('should render', () => {
    const { asFragment } = render(<Input label="test-label" icon={<>Icon</>} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
