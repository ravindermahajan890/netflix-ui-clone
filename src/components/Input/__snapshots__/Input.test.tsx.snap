// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: rgba(255, 255, 255, 0.7);
  font-family: Inter;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  color: white;
  font-size: 20px;
}

.emotion-0.Mui-focused {
  color: #000;
}

.emotion-0.Mui-disabled {
  color: rgba(255, 255, 255, 0.5);
}

.emotion-0.Mui-error {
  color: #DC3545;
}

.emotion-1 {
  font-family: Inter;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: #fff;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #ADB5BD;
  border-radius: 4px;
  width: 100%;
  padding: 16px;
  margin-top: 8px;
}

.emotion-1.Mui-disabled {
  color: rgba(255, 255, 255, 0.5);
  cursor: default;
}

.emotion-1 .MuiInputBase-input {
  font-size: 20px;
  font-weight: 300;
  padding: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: #fff;
  margin-right: 8px;
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.5;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(255, 255, 255, 0.5);
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

<div>
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated emotion-0"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart emotion-1"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart emotion-2"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
        Icon
      </div>
      <input
        class="MuiInputBase-input MuiInputBase-inputAdornedStart emotion-3"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
